webapp/src/features/adminAddIntersection/adminAddIntersectionSlice.tsx

```diff
@@ -0,0 +1,258 @@
+import { PayloadAction, createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+import { selectToken } from '../../generalSlices/userSlice'
+import EnvironmentVars from '../../EnvironmentVars'
+import apiHelper from '../../apis/api-helper'
+import { updateTableData as updateIntersectionTableData } from '../adminIntersectionTab/adminIntersectionTabSlice'
+import { RootState } from '../../store'
+import { AdminAddIntersectionForm } from './AdminAddIntersection'
+
+export type AdminIntersectionCreationInfo = {
+  organizations: string[]
+  rsus: string[]
+}
+
+export type AdminIntersectionKeyedCreationInfo = {
+  organizations: { id: number; name: string }[]
+  rsus: { id: number; name: string }[]
+}
+
+export type AdminIntersectionCreationBody = {
+  intersection_id: string
+  ref_pt: {
+    latitude: string
+    longitude: string
+  }
+  bbox?: {
+    latitude1: string
+    longitude1: string
+    latitude2: string
+    longitude2: string
+  }
+  intersection_name?: string
+  origin_ip?: string
+  organizations: string[]
+  rsus: string[]
+}
+
+const initialState = {
+  apiData: {} as AdminIntersectionKeyedCreationInfo,
+  selectedOrganizations: [] as AdminIntersectionKeyedCreationInfo['organizations'],
+  selectedRsus: [] as AdminIntersectionKeyedCreationInfo['rsus'],
+  submitAttempt: false,
+}
+
+export const convertApiJsonToKeyedFormat = (
+  apiJson: AdminIntersectionCreationInfo
+): AdminIntersectionKeyedCreationInfo => {
+  if (Object.keys(apiJson).length !== 0) {
+    let keyedApiJson = {} as AdminIntersectionKeyedCreationInfo
+
+    let data = []
+    for (let i = 0; i < apiJson['organizations'].length; i++) {
+      let value = apiJson['organizations'][i]
+      let temp = { id: i, name: value }
+      data.push(temp)
+    }
+    keyedApiJson.organizations = data
+
+    data = []
+    for (let i = 0; i < apiJson['rsus'].length; i++) {
+      let value = apiJson['rsus'][i]
+      let temp = { id: i, name: value?.replace('/32', '') }
+      data.push(temp)
+    }
+    keyedApiJson.rsus = data
+
+    return keyedApiJson
+  }
+}
+
+export const checkForm = (state: RootState['adminAddIntersection']) => {
+  if (state.value.selectedOrganizations.length === 0) {
+    return false
+  } else {
+    return true
+  }
+}
+
+export const updateJson = (
```

Comment:
We updated the other json conversion function name, but there's still some in here like this updateJson that we should rename and add descriptions to accordingly.

Reply:
Added docstring comments and renamed method